cli/crates/server/src/bridge/search.rs

```diff
@@ -0,0 +1,401 @@
+                        .to_vec(),
+                )
+                .expect("Id is a valid string"),
+            );
+        }
+
+        trace!(
+            "Found {} top documents:\n{:?}",
+            matching_records.len(),
+            matching_records
+        );
+        Ok(matching_records)
+    }
+
+    pub async fn build(pool: &SqlitePool, entity_type: &str, search_schema: &SearchSchema) -> Result<Index, ApiError> {
+        trace!("Building index for {entity_type} and schema:\n{search_schema:?}");
+
+        let mut schema_builder = Schema::builder();
+        let id_field = schema_builder.add_bytes_field("id", STORED);
+        let (fields, default_fields) = {
+            let mut all = vec![];
+            let mut defaults = vec![];
+            let mut search = vec![];
+
+            for field in &search_schema.fields {
+                let base_type = BASE_TYPE_RE.replace_all(&field.r#type, "");
+                search.push(SearchField {
+                    r#type: base_type.to_string(),
+                    ..field.clone()
+                });
+                let field = match base_type.borrow() {
+                    scalars::URL | scalars::EMAIL | scalars::PHONE => {
+                        schema_builder.add_text_field(&field.name, STRING)
+                    }
+                    scalars::STRING => schema_builder.add_text_field(&field.name, TEXT),
+                    scalars::DATE | scalars::DATE_TIME | scalars::TIMESTAMP => {
+                        schema_builder.add_date_field(&field.name, INDEXED)
+                    }
+                    scalars::INT => schema_builder.add_i64_field(&field.name, INDEXED),
+                    scalars::FLOAT => schema_builder.add_f64_field(&field.name, INDEXED),
+                    scalars::BOOL => schema_builder.add_bool_field(&field.name, INDEXED),
+                    scalars::IP => schema_builder.add_ip_addr_field(&field.name, INDEXED),
+                    unknown_type => {
+                        error!("Unknown field type: '{unknown_type}'");
+                        return Err(ApiError::ServerError);
+                    }
+                };
+                all.push(field);
+                match base_type.borrow() {
+                    scalars::STRING | scalars::URL | scalars::EMAIL | scalars::PHONE => defaults.push(field),
+                    _ => (),
+                };
+            }
+            (search.into_iter().zip(all).collect::<Vec<_>>(), defaults)
+        };
+        let schema = schema_builder.build();
+        let builder = tantivy::Index::builder()
+            .schema(schema.clone())
+            .settings(IndexSettings {
+                docstore_compression: Compressor::None,
+                ..Default::default()
+            });
+
+        let index = builder.open_or_create(RamDirectory::create())?;
+        let mut index_writer = index.writer_with_num_threads(1, 20_000_000)?;
+
+        let mut fut = sqlx::query_as(
```

Comment:
Could we add this to the SQL type?

Reply:
Which SQL type? There's only one I know of in api.

Reply:
Ah right, I'm used to SQL queries being formed there. Nevermind